src/main/java/bio/terra/service/dataset/DatasetDao.java

```diff
@@ -400,13 +393,10 @@ private DataAccessException getFaultToInsert(LockType lockType) {
    * unlock.
    *
    * @param dataset the dataset object to create
-   * @return the id of the new dataset
-   * @throws IOException
    * @throws InvalidDatasetException if a row already exists with this dataset name
    */
   @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
-  public void createAndLock(Dataset dataset, String flightId, AuthenticatedUserRequest userReq)
-      throws IOException {
+  public void createAndLock(Dataset dataset, String flightId) throws IOException {
```

Comment:
Oh wow, that's great that you could remove the userReq. Looks like we were just passing it through and not using it?
I suppose we're assuming that if you get to the Dao layer, that the permissions have been checked. I wish there was a non-tedious way to confirm that we're always check this, especially for the delete operation below.

Reply:
Correct, we were passing it through but it was not used.  This happened in a few similar places.  Fortunately, it looks like they were never used, and were only added in within the last year as an oversight:

https://github.com/DataBiosphere/jade-data-repo/commit/aba66fdf32e1eadf07d284e1d2b86d079c849bc1

Reply:
Ah, thanks for tracking down the original change here!